lexos/models/stats_model.py

```diff
@@ -159,9 +174,20 @@ def get_corpus_stats(self) -> CorpusStats:
         # Get the standard deviation of the file word counts.
         std_deviation = file_sizes.std(axis="index")
 
+        # Get quartile indexes
+        arr = np.array(file_sizes)
+        arr.sort()
+        length = len(arr)
+        if length == 1:
+            q1_index = 0
+            q3_index = 0
+        else:
+            q1_index = length * .25 + .5 - 1
+            q3_index = length * .75 + .5 - 1
+
         # Get the iqr of the file word counts.
-        first_quartile = file_sizes.quantile(0.25)
-        third_quartile = file_sizes.quantile(0.75)
+        first_quartile = self.__get_quartile__(index=q1_index, arr=arr)
```

Comment:
Does the performance of this function outrun numpy's performance? What was the reason to recreate the wheel?

Reply:
I agree, also it seems confusing to me how they get q1_index, what is the purpose of the if statement, which set both q1_index and q3_index to zero?
Does numpy quantile not handle it correctly?

Reply:
The numpy quantile calculates the quartiles in a different way than how plotly calculates them and the results are slightly different and Mark wanted them to be the same results. Setting both q1_index and q3_index to zero is for when only one document is put through stats. If they are not set to zero, q1_index is a negative number and it breaks.

Reply:
Adding these explanation in the comments will be helpful for future developers.